src/basic/log.h

```diff
@@ -367,3 +368,41 @@ int log_syntax_invalid_utf8_internal(
 #define DEBUG_LOGGING _unlikely_(log_get_max_level() >= LOG_DEBUG)
 
 void log_setup(void);
+
+typedef struct ErrorRateLimit {
+        int error;
+        RateLimit ratelimit;
+} ErrorRateLimit;
+
+void log_dropped_errors(size_t num_dropped_errors, int level);
+
+bool error_is_ratelimited(int error, ErrorRateLimit *error_ratelimit);
+
+#define log_ratelimit_internal(_level, _error, _file, _line, _func, ...)        \
+({                                                                              \
+        int _log_ratelimit_error = (_error);                                    \
+        static ErrorRateLimit error_ratelimit = {                               \
+                .ratelimit = {                                                  \
+                        .interval = (1 * USEC_PER_SEC),                         \
+                        .burst = 1                                              \
+                },                                                              \
+        };                                                                      \
+        static size_t num_dropped_errors;                                       \
```

Comment:
hmm, so the RateLimit object already counts hits. it currently stops doing so once the limit is reached though. I think we should change that hoewver, and then continue counting. That way we don't have to count hits while limited separately (like you do here), but instead get the count of dropped messages simply by subtracting the limit from `ratelimit->num`.

Reply:
Thanks for the suggestion! I implemented it. It does simplify things :)